### PR TITLE
Move K8s rules under YAML language

### PR DESCRIPTION
**Pull request by mschwager**
I had to re-arrange the test quite a bit because the YAML `pre-commit` hook is now checking it, and it was complaining about:

```
expected a single document in the stream
  in "yaml/kubernetes/best-practice/no-fractional-cpu-limits.test.yaml", line 1, column 1
but found another document
  in "yaml/kubernetes/best-practice/no-fractional-cpu-limits.test.yaml", line 45, column 1
```

But it should be good to go now :+1:

### CODE REVIEW

File: yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml [hunk 1]
```diff
@@ -16,10 +16,4 @@ rules:
     even though most nodes can handle processes
     alternating between using 100% of the CPU.
   severity: WARNING
-  paths:
```
Comment (mschwager) [inline yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml:19]: Is `.k8s` a common extension for these configs?

The `.yml` and `.yaml` should be covered after https://github.com/returntocorp/semgrep/pull/2877.

Comment (emjin) [reply yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml:19]: I think it is; k8s is kubernetes configs I believe

Comment (emjin) [reply yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml:19]: Related issue: https://github.com/returntocorp/semgrep/issues/2837 (might not be relevant, I'm a little bit confused about the root cause of this issue)

File: yaml/kubernetes/best-practice/no-fractional-cpu-limits.test.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            exec:
+              command: [chmod, '0777', /data/i.underyx.me]
+        livenessProbe:
+          tcpSocket:
+            port: 22
+          initialDelaySeconds: 30
+          periodSeconds: 30
+        resources:
+          limits:
+              # ok: kubernetes.best-practice.no-fractional-cpu-limits
```
Comment (underyx) [inline yaml/kubernetes/best-practice/no-fractional-cpu-limits.test.yaml]:
```suggestion
            # ok: kubernetes.best-practice.no-fractional-cpu-limits
```

File: yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml (hunk 1 above)
Comment (underyx) [reply yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml:19]: I've never seen `.k8s` used in the wild so I'd remove it.

Comment (underyx) [reply yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml:19]: @sim1 have you ever seen manifests use a `.k8s` extension?

**Review by emjin — APPROVED** [review]

File: yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml (hunk 1 above)
Comment (iakat) [reply yaml/kubernetes/best-practice/no-fractional-cpu-limits.yaml:19]: no, this is the first I hear of a .k8s file extension @underyx

File: yaml/kubernetes/best-practice/no-fractional-cpu-limits.test.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        - name: plex-transcode
+          mountPath: /transcode
+        livenessProbe:
+          tcpSocket:
+            port: 32400
+          initialDelaySeconds: 30
+          periodSeconds: 30
+        resources:
+          limits:
+              # ok: kubernetes.best-practice.no-fractional-cpu-limits
```
Comment (mschwager) [inline yaml/kubernetes/best-practice/no-fractional-cpu-limits.test.yaml]:
```suggestion
            # ok: kubernetes.best-practice.no-fractional-cpu-limits
```

**Pull request merged**